modules/apps/app-builder/app-builder-test/src/testIntegration/java/com/liferay/app/builder/deploy/test/AppDeployerTest.java

```diff
@@ -0,0 +1,217 @@
+import com.liferay.app.builder.service.AppBuilderAppLocalService;
+import com.liferay.app.builder.service.AppBuilderAppVersionLocalService;
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.test.ReflectionTestUtil;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DataGuard;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.ServiceRegistration;
+
+/**
+ * @author Rebeca Silva
+ * @author Alejo Ceballos
+ */
+@DataGuard(scope = DataGuard.Scope.METHOD)
+@RunWith(Arquillian.class)
+public class AppDeployerTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testProductMenuDeployerDeployInApplicationsMenu()
+		throws Exception {
+
+		AppBuilderApp appBuilderApp = _createAppBuilderApp();
+
+		JSONObject settingsJSONObject = JSONFactoryUtil.createJSONObject(
+			JSONUtil.put(
+				"scope", JSONUtil.put("applications_menu.applications")
+			).toJSONString());
+
+		_appBuilderAppDeploymentLocalService.addAppBuilderAppDeployment(
+			appBuilderApp.getAppBuilderAppId(),
+			settingsJSONObject.toJSONString(), "productMenu");
+
+		_deployAndUndeployAppBuilderApp(
+			appBuilderApp, "applications_menu.applications");
+	}
+
+	@Test
+	public void testProductMenuDeployerDeployInApplicationsMenuAndSites()
+		throws Exception {
+
+		AppBuilderApp appBuilderApp = _createAppBuilderApp();
+
+		JSONArray jsonArray = JSONFactoryUtil.createJSONArray();
+
+		JSONObject siteAndApplicationsJSONObject =
+			JSONFactoryUtil.createJSONObject(
+				JSONUtil.put(
+					"scope",
+					jsonArray.put(
+						"applications_menu.applications"
+					).put(
+						"site_administration.content"
+					)
+				).put(
+					"siteIds", JSONUtil.put(-1)
+				).toJSONString());
+
+		_appBuilderAppDeploymentLocalService.addAppBuilderAppDeployment(
+			appBuilderApp.getAppBuilderAppId(),
+			siteAndApplicationsJSONObject.toJSONString(), "productMenu");
+
+		_deployAndUndeployAppBuilderApp(
+			appBuilderApp, "applications_menu.applications",
```

Comment:
I know we don't usually create constants, but just bringing up this point (about `applications_menu.applications` and `site_administration.content`)... what do you think?

Reply:
I'm not sure if its' necessary here. let's see what brian says